formver.compiler-plugin/uniqueness/src/org/jetbrains/kotlin/formver/uniqueness/UniqueCheckVisitor.kt

```diff
@@ -86,11 +86,16 @@ object UniqueCheckVisitor : FirVisitor<Pair<UniqueLevel, UniquePathContext?>, Un
         val symbol = functionCall.toResolvedCallableSymbol()
         val params = (symbol as FirFunctionSymbol<*>).fir.valueParameters
         val requiredUniqueLevels = params.map { data.resolveUniqueAnnotation(it) }
-        // Skip merge of context for now
+        val paramBorrowingLevels = params.map { data.resolveBorrowingAnnotation(it) }
+
         val arguments = functionCall.arguments
         arguments.forEachIndexed { index, argument ->
             val requiredUnique = requiredUniqueLevels[index]
+            val paramBorrowing = paramBorrowingLevels[index]
             val (argumentUnique, trie) = argument.accept(this, data)
+            val argumentBorrowing =
+                trie?.localVariable?.let { data.resolveBorrowingAnnotation(it) } ?: BorrowingLevel.Owned
```

Comment:
It feels pretty ugly to handle this here, you need to really get into the details of the trie.  Maybe there's a nicer interface to extract?